### PR TITLE
feat(sas backends): Offboard a team from a backend if it's removed from backend list

### PR DESCRIPTION
**Pull request by rp926463-arch**
…verage

- Add handleRemovedBackends() to detect and offboard from removed backends
- Add detectRemovedBackends() to compare spec vs status for removed backends
- Add offboardFromRemovedBackends() to perform cleanup using existing DeleteTeamByID
- Add comprehensive unit tests covering all error paths and edge cases
- Ensure graceful handling of cache misses, client creation errors, and transformation failures
- All 17 controller tests passing with 30.2% coverage
- Zero linting issues - production ready

Resolves: DATA-3526

# Changes

### Dependencies

- N/A

---

### Performance Impact

- N/A

---

### Deploy Steps

1. N/A

### Prerequisites

- N/A

### Post-Deployment Monitoring

- N/A

### Rollback Plan

- N/A

---

## ⚠️ Breaking Changes

- [ ] This PR contains breaking changes
- [ ] Migration guide provided (if applicable)

**Details:**

- N/A

---

## ⚙️ Configuration Changes

- N/A

---

## ✅ Developer Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added positive and negative tests that prove my fix is effective or that my feature works
- [x] Relevant documentation (README, tech specs, etc.) has been added or updated
- [x] All CI/CD checks are passing

### CODE REVIEW

File: internal/controller/group_controller.go [hunk 1]
```diff
@@ -892,3 +898,129 @@ func isGroupCRHasDependants(backends []usernautdevv1alpha1.Backend, dependsOn co
...
+				backendLoggerInfo.Info("offboarding: team not found in cache, skipping deletion")
+			}
+		} else {
+			backendLoggerInfo.Info("offboarding: cache entry not found, skipping deletion")
+		}
+	}
+
+	r.log.Info("completed offboarding process for all removed backends")
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (rp926463-arch) [reply internal/controller/group_controller.go]: This is addressed

File: internal/controller/group_controller_test.go [hunk 2]
```diff
@@ -284,5 +285,462 @@ var _ = Describe("Group Controller", func() {
...
+				By("calling offboardFromRemovedBackends")
+				err = reconciler.offboardFromRemovedBackends(ctx, groupCR, removedBackends)
+
+				By("verifying error occurs due to client creation failure")
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("missing required connection parameters"))
+			})
+
+			It("should handle cache entry not found gracefully", func() {
+				By("testing that empty cache is handled gracefully without client creation")
```
[the remark this answers is not in the document]
Comment (rp926463-arch) [reply internal/controller/group_controller_test.go]: This is addressed

File: internal/controller/group_controller.go [hunk 3]
```diff
@@ -892,3 +898,133 @@ func isGroupCRHasDependants(backends []usernautdevv1alpha1.Backend, dependsOn co
...
+			removedBackends = append(removedBackends, statusBackend)
+		}
+	}
+
+	return removedBackends
+}
+
+// offboardFromRemovedBackends performs the actual cleanup for removed backends
+// This reuses the existing deleteBackendsTeam logic but for specific backends only
+func (r *GroupReconciler) offboardFromRemovedBackends(ctx context.Context, groupCR *usernautdevv1alpha1.Group, removedBackends []usernautdevv1alpha1.BackendStatus) error {
```
[the remark this answers is not in the document]
Comment (rp926463-arch) [reply internal/controller/group_controller.go]: This is addressed, if multiple backends are removed and some fail to clean up, the function will still attempt to clean up all the others and provide complete visibility into what succeeded and what failed.

File: internal/controller/group_controller_test.go [hunk 4]
```diff
@@ -284,5 +285,462 @@ var _ = Describe("Group Controller", func() {
...
+				}
+
+				By("calling handleRemovedBackends")
+				err = reconciler.handleRemovedBackends(ctx, groupNoRemoval)
+
+				By("verifying no error occurs when no backends need removal")
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			// DATA-3526: Unit tests for offboardFromRemovedBackends function error scenarios
```
[the remark this answers is not in the document]
Comment (rp926463-arch) [reply internal/controller/group_controller_test.go]: Above tests added.

File: internal/controller/group_controller.go [hunk 5]
```diff
@@ -892,3 +898,133 @@ func isGroupCRHasDependants(backends []usernautdevv1alpha1.Backend, dependsOn co
...
+
+// detectRemovedBackends compares the current status against the spec to find removed backends
+func (r *GroupReconciler) detectRemovedBackends(groupCR *usernautdevv1alpha1.Group) []usernautdevv1alpha1.BackendStatus {
+	var removedBackends []usernautdevv1alpha1.BackendStatus
+
+	// Create a map of current backends for quick lookup
+	currentBackends := make(map[string]bool)
+	for _, backend := range groupCR.Spec.Backends {
+		// Use name_type as key for uniqueness
+		key := backend.Name + "_" + backend.Type
```
[the remark this answers is not in the document]
Comment (rp926463-arch) [reply internal/controller/group_controller.go]: This is addressed
